legs_test.go

```diff
@@ -197,11 +200,21 @@ func TestSetAndFilterPeerPolicy(t *testing.T) {
 	np := basicnode.Prototype__Any{}
 	nb := np.NewBuilder()
 	ma, _ := nb.BeginMap(2)
-	ma.AssembleKey().AssignString("hey")
-	ma.AssembleValue().AssignString("it works!")
-	ma.AssembleKey().AssignString("yes")
-	ma.AssembleValue().AssignBool(true)
-	ma.Finish()
+	if err = ma.AssembleKey().AssignString("hey"); err != nil {
```

Comment:
It does not provide any additional functionality, only syntax that may be preferable to some.  I would prefer that be done in a separate PR if you feel it would be a useful improvement.